### ISSUE TITLE
workers script

### ISSUE THREAD

**Issue by afletch** [rq/rq#1066]
Following the docs at: https://python-rq.org/docs/workers/#performance-notes

This suggests that common setup can be done before the fork, however this does not appear to work.

Importing modules into the worker script does not make this available in the job, as the docs suggest it should

**Comment by mlissner** [comment]
I was just going to file an issue about this too. The worker script that's provided there is pretty painfully incomplete. Ideally this script would support all of the arguments supported by the regular worker. It's frustrating to see a section that says, "If you want better performance, do this thing, but plan on breaking the rest of your set up."

I think this is particularly painful because the worker is usually wrapped up by docker or supervisor or something, so when you get to performance tuning (at the end of your project), you have to go back and re-do supervisor or whatever.

**Comment by afletch** [comment]
@mlissner Did you manage to get any of your common setup done in the worker before the fork and have this available in the job?

**Comment by mlissner** [comment]
Nope! I gave up and decided to look into it after I got retries figured out (still working on that). Happy to join forces if it'd be helpful, but it seemed like doing this was premature optimization until I knew how big an impact it'd have.

**Comment by afletch** [comment]
@selwin Would you be able to offer any pointers here?

**Comment by Houd1ny** [comment]
I have the same question.

**Comment by afletch** [comment]
I never managed to get to the bottom of this. Instead I used a custom worker class.

**Comment by Houd1ny** [comment]
Guys, I think I understood how this should work
https://github.com/rq/rq/issues/1088

**Comment by lowercase00** [comment]
@Houd1ny experiments mention:

(1) load the module with the tasks (instead of the libraries)
(2) use the `SimpleWorker`, which doesn't fork

We should update the docs to make this clearer.
